From 8ec77cd15997749f51912e75bd47fad032cd3afe Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Wed, 12 Feb 2014 16:05:30 -0600
Subject: [PATCH 2/2] ARM: OMAP2+: omap_hwmod: Add context ops to am43xx
 soc_ops

AM43XX contains PRM_*_CONTEXT registers like OMAP4 did to track loss
of context in modules so utilize the same soc_ops present for OMAP4.

Certain drivers use the context loss count to determine if context was
truly lost during a suspend operation to decide whether or not a
context restore is truly necessary. Without this, some drivers may
choose not to restore context when they have actually lost it and fail
to properly resume.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-omap2/omap_hwmod.c b/arch/arm/mach-omap2/omap_hwmod.c
index 1f766fd..c03c54d 100644
--- a/arch/arm/mach-omap2/omap_hwmod.c
+++ b/arch/arm/mach-omap2/omap_hwmod.c
@@ -4296,6 +4296,8 @@ void __init omap_hwmod_init(void)
 		soc_ops.deassert_hardreset = _omap4_deassert_hardreset;
 		soc_ops.is_hardreset_asserted = _omap4_is_hardreset_asserted;
 		soc_ops.init_clkdm = _init_clkdm;
+		soc_ops.update_context_lost = _omap4_update_context_lost;
+		soc_ops.get_context_lost = _omap4_get_context_lost;
 	} else if (soc_is_am33xx()) {
 		soc_ops.enable_module = _am33xx_enable_module;
 		soc_ops.disable_module = _am33xx_disable_module;
-- 
1.7.9.5

